src/Phpro/SoapClient/BridgeBundle/Resources/views/Toolbar/index.html.twig

```diff
@@ -1,81 +1,194 @@
+            <div class="metric">
+                <span class="value">{{ collector.calls | length }}</span>
+                <span class="label">Soap Requests</span>
+            </div>
+
+            <div class="metric">
+                <span class="value">{{ '%0.2f'|format(collector.totalTiming) }} ms</span>
+                <span class="label">Total Time</span>
+            </div>
+        </div>
+    {% endif %}
+
+    <h2>Requests</h2>
+
+    {% if collector.calls is empty %}
+        <div class="empty">
+            <p>No Soap requests were performed.</p>
+        </div>
+    {% else %}
+        <table class="alt soap-requests-table">
+            <thead>
             <tr>
-                <th>Response Object</th>
-                <td>{{ dump(call.response) }}</td>
+                <th class="nowrap">#</th>
+                <th class="nowrap">Time</th>
+                <th style="width: 100%;">Query</th>
             </tr>
+            </thead>
+            <tbody>
+            {% for request in collector.calls %}
+                <tr>
+                    <td class="nowrap">{{ loop.index }}</td>
+                    <td class="nowrap">{{ '%0.2f'|format(request.timing) }} ms </td>
+                    <td>
+                        <div>{{ request.clientName }}::{{ request.method }}</div>
+                        <a href="#" class="btn btn-link text-small sf-toggle" data-toggle-selector="#soap-request-headers-{{ loop.index }}" data-toggle-alt-content="Hide headers"{{ profiler_markup_version == 1 ? 'onclick="return toggle(this);"' }}>
+                            Show headers
+                        </a>
+
+                        <div id="soap-request-headers-{{ loop.index }}" class="hidden sf-toggle-content"><pre>{{ request.requestHeaders }}</pre></div>
+                        
+                        <a href="#" class="btn btn-link text-small sf-toggle" data-toggle-selector="#soap-request-body-{{ loop.index }}" data-toggle-alt-content="Hide body"{{ profiler_markup_version == 1 ? 'onclick="return toggle(this);"' }}>
+                            Show body
+                        </a>
+                        
+                        <div id="soap-request-body-{{ loop.index }}" class="hidden sf-toggle-content"><pre>{{ request.requestBody }}</pre></div>
+
+                        <a href="#" class="btn btn-link text-small sf-toggle" data-toggle-initial="display" data-toggle-selector="#soap-response-{{ loop.index }}" data-toggle-alt-content="Show response"{{ profiler_markup_version == 1 ? 'onclick="return toggle(this);"' }}>
+                            Hide response
+                        </a>
+                        
+                        <div id="soap-response-{{ loop.index }}" class="sf-toggle-content soap-response-container">
+                            <div style="margin-bottom: 15px;"><pre>{{ request.responseHeaders }}</pre></div>
+                            <div>{{ request.responseBody|nl2br }}</div>
```

Comment:
All requests and responses are ran trough DOMDocument for formatting in the SoapCall class.